platform-validation/scripts/compatibility.sh

```diff
@@ -393,6 +393,39 @@ perform_capability_check_container(){
     wrap_pass "capability_check_container" "Pass"
 }
 
+check_shared_library_dependency(){
+    wrap_debug "Checking shared library dependency for aziot-edged and aziot-identityd"
+
+    # check dependencies 
+    ret=$(need_cmd ldconfig) && $(need_cmd grep)
+    if [ $? != 0 ]; then 
+        wrap_warning "Skipping check_shared_library_dependency" 
+        return
+    fi
+
+    # set the crucial libaries in the variable
+    # IOTEDGE_COMMON_SHARED_LIBRARIES is for both aziot-edged and aziot-identityd 
+    IOTEDGE_COMMON_LIBRARIES="libssl.so.1.1 libcrypto.so.1.1 libdl.so.2 librt.so.1 libpthread.so.0 libc.so.6 libm.so.6 ld-linux-x86-64.so.2 libgcc_s.so.1"
+    for lib in $IOTEDGE_COMMON_LIBRARIES
+    do
+        ret=$(ldconfig -p | grep $lib)
+        if [ -z "$ret" ]; then 
+            case $lib in
+                "libssl.so.1.1" | "libcrypto.so.1.1") wrap_warning "$lib is missing. Please install openssl and libssl-dev"
```

Comment:
Are "openssl" and "libssl-dev" the names of Ubuntu-specific packages, and the names might be different on other platforms?

Reply:
Yes, they may, we should refrain from explicitly specifying a package name. Maybe we can recommend simply checking their distro's package manager to install the libraries

Reply:
I feel give some name is good to get started instead of providing a vague message to just install it? What do you guys think. I was going for _Please install openssl and libssl-dev for your OS distribution._